File: presentation/src/main/java/com/khalturin/dmitriy/presentation/viewmodel/bookmark/BookmarksViewModel.java
package com.khalturin.dmitriy.presentation.viewmodel.bookmark;

import android.databinding.ObservableField;

import com.khalturin.dmitriy.presentation.binding.BindingTransformer;
import com.khalturin.dmitriy.presentation.binding.recycler.RecyclerManager;
import com.khalturin.dmitriy.presentation.binding.recycler.adapter.BindingRecyclerAdapter;

import java.util.List;

import io.reactivex.Observable;

import static com.khalturin.dmitriy.presentation.viewmodel.DefaultFieldValue.EMPTY_STRING;
import static java.util.concurrent.TimeUnit.MILLISECONDS;

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 03.04.18 9:58.
 */

public class BookmarksViewModel {

//==================================================================================================
//    Binding ViewModel Members
//==================================================================================================

  public ObservableField<RecyclerManager> recyclerManager = new ObservableField<>();
  public ObservableField<String> searchText = new ObservableField<>(EMPTY_STRING);

//==================================================================================================
//    Class Methods
//==================================================================================================

  public void setBookmarksItems(List<RssFeedViewModel> items){
    getAdapter().setItems(items);
  }

  @SuppressWarnings("unchecked")
  private BindingRecyclerAdapter<RssFeedViewModel> getAdapter(){
    return (BindingRecyclerAdapter<RssFeedViewModel>) recyclerManager.get().getAdapter();
  }

  public Observable<String> getOnSearchChange(){
    int timeout = 1000;

    return BindingTransformer.toObservable(searchText)
      .debounce(timeout, MILLISECONDS);
  }

}

File: rssreader/src/main/kotlin/com/khalturin/dmitriy/rssreader/view/viewmodel/BookmarksViewModel.kt
package com.khalturin.dmitriy.rssreader.view.viewmodel

import android.databinding.ObservableField
import android.view.View
import com.khalturin.dmitriy.library.recyclerview.RecyclerViewManager

/**
 * Created by Dmitriy Khalturin <dmitry.halturin.86@gmail.com>
 * for android-rss-reader on 31.10.18 1:00.
 */
class BookmarksViewModel {

  val mRecyclerViewManager = ObservableField<RecyclerViewManager>()

  val mRssUrl = ObservableField<String>()

  fun loadRssFeed(view: View) {}

}
